src/service-catalog-appregistry-stack.ts

```diff
@@ -0,0 +1,83 @@
+import { OUTPUT_SERVICE_CATALOG_APPREGISTRY_APPLICATION_ARN } from './common/constant';
+import { SolutionInfo } from './common/solution-info';
+
+export interface ServiceCatalogAppRegistryProps extends StackProps {
+}
+
+export class ServiceCatalogAppregistryStack extends Stack {
+  constructor(scope: Construct, id: string, props?: ServiceCatalogAppRegistryProps) {
+    super(scope, id, props);
+
+    const featureName = 'AppRegistry';
+    this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-appreg ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
+
+    const projectIdParam = new CfnParameter(this, 'ProjectId', {
+      description: 'Project Id',
+      type: 'String',
+      default: '',
+    });
+
+    const serviceAvailableRegion = new CfnCondition(this, 'ServiceCatalogAvailableRegion', {
+      expression: Fn.conditionOr(
+        Fn.conditionEquals(Aws.REGION, 'us-east-1'),
+        Fn.conditionEquals(Aws.REGION, 'us-east-2'),
+        Fn.conditionEquals(Aws.REGION, 'us-west-1'),
+        Fn.conditionEquals(Aws.REGION, 'us-west-2'),
+        Fn.conditionEquals(Aws.REGION, 'ap-south-1'),
+        Fn.conditionEquals(Aws.REGION, 'ap-northeast-1'),
+        Fn.conditionEquals(Aws.REGION, 'ap-northeast-2'),
+        Fn.conditionEquals(Aws.REGION, 'ap-northeast-3'),
+        Fn.conditionEquals(Aws.REGION, 'ap-southeast-1'),
+        Fn.conditionEquals(Aws.REGION, 'ap-southeast-2'),
+        Fn.conditionEquals(Aws.REGION, 'ca-central-1'),
+        Fn.conditionEquals(Aws.REGION, 'eu-central-1'),
+        Fn.conditionEquals(Aws.REGION, 'eu-west-1'),
+        Fn.conditionEquals(Aws.REGION, 'eu-west-2'),
+        Fn.conditionEquals(Aws.REGION, 'eu-west-3'),
+        Fn.conditionEquals(Aws.REGION, 'eu-north-1'),
+        Fn.conditionEquals(Aws.REGION, 'sa-east-1'),
+      ),
+    });
+
+    const application = new Application(this, 'ServiceCatalogApplication', {
+      applicationName: Fn.join('-', [
+        'clickstream-analytics',
+        projectIdParam.valueAsString,
+        Aws.REGION,
+        Aws.ACCOUNT_ID,
+      ]),
+      description: `Catalog Service AppRegistry application for Clickstream Analytics project: ${projectIdParam.valueAsString}`,
+    });
+    // Add condition for region validation
+    (application.node.defaultChild as CfnResource).cfnOptions.condition = serviceAvailableRegion;
+
+    // Add tags for AppRegistry application
+    Tags.of(application).add('Solutions:SolutionID', SolutionInfo.SOLUTION_ID);
+    Tags.of(application).add('Solutions:SolutionName', SolutionInfo.SOLUTION_NAME);
+    Tags.of(application).add('Solutions:SolutionVersion', SolutionInfo.SOLUTION_VERSION);
+    Tags.of(application).add('Solutions:ApplicationType', 'AWS-Solutions');
+
+    new CfnOutput(this, OUTPUT_SERVICE_CATALOG_APPREGISTRY_APPLICATION_ARN, {
+      description: 'Service Catalog AppRegistry Application Arn',
+      value: application ? application.applicationArn : '',
```

Comment:
should be `application.applicationArn`, because the application always is not empty